core/src/main/java/nl/nn/adapterframework/narayana/DataSourceXAResourceRecoveryHelper.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nl.nn.adapterframework.narayana;
+
+import java.sql.SQLException;
+
+import javax.sql.XAConnection;
+import javax.sql.XADataSource;
+import javax.transaction.xa.XAException;
+import javax.transaction.xa.XAResource;
+import javax.transaction.xa.Xid;
+
+import com.arjuna.ats.jta.recovery.XAResourceRecoveryHelper;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.util.Assert;
+
+/**
+ * XAResourceRecoveryHelper implementation which gets XIDs, which needs to be recovered,
+ * from the database.
+ *
+ * @author Gytis Trikleris
+ * @since 1.4.0
```

Comment:
Is deze code 1:1 overgenomen? Het lijkt me goed om in het commentaar een verwijzing naar het origineel op te nemen